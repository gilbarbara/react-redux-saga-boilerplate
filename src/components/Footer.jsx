import React from 'react';
import styled from 'styled-components';

import { Container, Flex } from 'styled-minimal';

const StyledFooter = styled.footer`
  border-top: 0.1rem solid #ddd;
`;

const Footer = () => (
  <StyledFooter>
    <Container py={3}>
      <Flex justifyContent="space-between">
        <iframe
          title="GitHub Stars"
          src="https://ghbtns.com/github-btn.html?user=gilbarbara&repo=react-redux-saga-boilerplate&type=star&count=true"
          frameBorder="0"
          scrolling="0"
          width="110px"
          height="20px"
        />
        <iframe
          title="GitHub Follow"
          src="https://ghbtns.com/github-btn.html?user=gilbarbara&type=follow&count=true"
          frameBorder="0"
          scrolling="0"
          width="130px"
          height="20px"
        />
      </Flex>
    </Container>
  </StyledFooter>
);

export default Footer;
