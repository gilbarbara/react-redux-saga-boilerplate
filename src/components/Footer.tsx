import React from 'react';
import { Box, Container, Flex } from 'styled-minimal';

function Footer() {
  return (
    <Box as="footer" borderTop="0.1rem solid #ddd">
      <Container py={3}>
        <Flex justifyContent="space-between">
          <iframe
            title="GitHub Stars"
            src="https://ghbtns.com/github-btn.html?user=gilbarbara&repo=react-redux-saga-boilerplate&type=star&count=true"
            frameBorder="0"
            scrolling="0"
            width="110px"
            height="20px"
          />
          <iframe
            title="GitHub Follow"
            src="https://ghbtns.com/github-btn.html?user=gilbarbara&type=follow&count=true"
            frameBorder="0"
            scrolling="0"
            width="130px"
            height="20px"
          />
        </Flex>
      </Container>
    </Box>
  );
}

export default Footer;
