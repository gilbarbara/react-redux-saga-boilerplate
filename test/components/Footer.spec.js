import React from 'react';
import { mount } from 'enzyme';

import Footer from 'components/Footer';

function setup() {
  return mount(<Footer />);
}

describe('Footer', () => {
  const wrapper = setup();

  it('should be a StatelessComponent', () => {
    expect(wrapper.instance()).toBeNull();
  });

  it('should render properly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
