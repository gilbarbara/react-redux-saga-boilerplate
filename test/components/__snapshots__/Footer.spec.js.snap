// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render properly 1`] = `
.c1 {
  margin-left: auto;
  margin-right: auto;
  padding-left: 16px;
  padding-right: 16px;
  max-width: 1440px;
  position: relative;
  width: 100%;
  padding-top: 16px;
  padding-bottom: 16px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 {
  border-top: 0.1rem solid #ddd;
}

@media (min-width:768px) {
  .c1 {
    padding-left: 32px;
    padding-right: 32px;
  }
}

<StyledFooter>
  <footer
    className="c0"
  >
    <StyledContainer
      py={3}
      verticalPadding={false}
    >
      <div
        className="c1"
      >
        <StyledFlex
          justifyContent="space-between"
        >
          <div
            className="c2"
          >
            <iframe
              frameBorder="0"
              height="20px"
              scrolling="0"
              src="https://ghbtns.com/github-btn.html?user=gilbarbara&repo=react-redux-saga-boilerplate&type=star&count=true"
              title="GitHub Stars"
              width="110px"
            />
            <iframe
              frameBorder="0"
              height="20px"
              scrolling="0"
              src="https://ghbtns.com/github-btn.html?user=gilbarbara&type=follow&count=true"
              title="GitHub Follow"
              width="130px"
            />
          </div>
        </StyledFlex>
      </div>
    </StyledContainer>
  </footer>
</StyledFooter>
`;
